ipfs-api/src/client/internal.rs

```diff
@@ -852,26 +852,31 @@ impl IpfsClient {
     /// ```
     ///
     #[inline]
-    pub async fn dag_get(&self, path: &str) -> Result<response::DagGetResponse, Error> {
-        self.request(request::DagGet { path }, None).await
-    }
-
-    // TODO /dag routes are experimental, and there isn't a whole lot of
-    // documentation available for how this route works.
-    //
-    // /// Add a DAG node to Ipfs.
-    // ///
-    // #[inline]
-    // pub fn dag_put<R>(&self, data: R) -> AsyncResponse<response::DagPutResponse>
-    // where
-    //     R: 'static + Read + Send,
-    // {
-    //     let mut form = multipart::Form::default();
-    //
-    //     form.add_reader("arg", data);
-    //
-    //     self.request(&request::DagPut, Some(form))
-    // }
+    pub async fn dag_get(&self, path: &str) -> Result<String, Error> {
```

Comment:
Can you remove the `response::DagGetResponse` type? It shouldn't be necessary if this API route returns a string.

Reply:
Also, I don't know too much about the DAG API, so I'm wondering if it's guaranteed to return a UTF8-encoded string? If not, it might be better to mimic how `cat` is implemented to handle generic binary outputs.

Reply:
> Can you remove the `response::DagGetResponse` type? It shouldn't be necessary if this API route returns a string.

I left it because UnixFS is the most common node type but I can remove it if you want, the user would have to write the struct.

> Also, I don't know too much about the DAG API, so I'm wondering if it's guaranteed to return a UTF8-encoded string? If not, it might be better to mimic how `cat` is implemented to handle generic binary outputs.

I was using JSON so I assumed UTF-8 but your right.

Reply:
> I left it because UnixFS is the most common node type but I can remove it if you want, the user would have to write the struct.

Hmm.. ok I think it makes sense to keep it then